hardware/detect.py

```diff
@@ -439,26 +446,23 @@ def find_element(xml, xml_spec, sys_subtype,
     else:
         status, output = cmd('lshw -xml')
     if status == 0:
+        mobo_id = ''
+        nic_id = ''
         xml = ET.fromstring(output)
         find_element(xml, "./node/serial", 'serial')
         find_element(xml, "./node/product", 'name')
         find_element(xml, "./node/vendor", 'vendor')
         find_element(xml, "./node/version", 'version')
         uuid = get_uuid()
+
         if uuid:
-            # If we have an uuid, let's manage a quirk list of stupid
-            # serial numbers TYAN or Supermicro are known to provide
-            # dirty serial numbers In that case, let's use the uuid
-            # instead
-            for i in hw_lst:
-                if 'system' in i[0] and 'product' in i[1] and 'serial' in i[2]:
-                    # Does the current serial number is part of the quirk list
-                    if i[3] in ['0123456789']:
-                        # Let's delete the stupid SN and use the UUID instead
-                        hw_lst.remove(i)
-                        hw_lst.append(('system', 'product', 'serial', uuid))
-                        break
-            hw_lst.append(('system', 'product', 'uuid', uuid))
+            # If we have an uuid, we shall check if it's part of a
+            # known list of broken uuid
+            # If so let's delete the uuid instead of reporting a stupid thing
+            if uuid in ['Not']:
```

Comment:
That is coming from a real case where the uuid is "Not" .... yes .... that
does exists....

2015-02-20 20:17 GMT+01:00 Emilien Macchi <notifications@github.com>:

> In hardware/detect.py
> <https://github.com/enovance/hardware/pull/5#discussion_r25094875>:
>
> > -            # serial numbers TYAN or Supermicro are known to provide
> > -            # dirty serial numbers In that case, let's use the uuid
> > -            # instead
> > -            for i in hw_lst:
> > -                if 'system' in i[0] and 'product' in i[1] and 'serial' in i[2]:
> > -                    # Does the current serial number is part of the quirk list
> > -                    if i[3] in ['0123456789']:
> > -                        # Let's delete the stupid SN and use the UUID instead
> > -                        hw_lst.remove(i)
> > -                        hw_lst.append(('system', 'product', 'serial', uuid))
> > -                        break
> > -            hw_lst.append(('system', 'product', 'uuid', uuid))
> > +            # If we have an uuid, we shall check if it's part of a
> > +            # known list of broken uuid
> > +            # If so let's delete the uuid instead of reporting a stupid thing
> > +            if uuid in ['Not']:
>
> Why "Not" is a specific case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/enovance/hardware/pull/5/files#r25094875>.
>